community/cypher/acceptance/src/test/scala/org/neo4j/internal/cypher/acceptance/FunctionsAcceptanceTest.scala

```diff
@@ -59,6 +67,14 @@ class FunctionsAcceptanceTest extends ExecutionEngineFunSuite with NewPlannerTes
     result should equal(4.0)
   }
 
+  test("toFloat should handle mixed number types") {
```

Comment:
It seems like `CTString` is also a possible combo with floats and ints. Perhaps add some tests with that one too?

What about negative acceptance tests? Ie, testing that combinations with incompatible types throws the expected error?